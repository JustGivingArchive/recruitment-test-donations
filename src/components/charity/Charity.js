import React from 'react';

const Charity = ({ info }) => {
  const { charityName, charityDescription, charityWebsite } = info;
  return (
    <a href={charityWebsite}>
    <div className="main_container hoverable">
      <h2 className="subtitle">{charityName}</h2>
      <p className="charity_description">{charityDescription}</p>
    </div>
    </a>

  );
};

export default Charity;
