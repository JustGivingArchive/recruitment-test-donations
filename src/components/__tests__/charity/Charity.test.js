import React from 'react';
import renderer from 'react-test-renderer';
import Charity from '../../charity/Charity.js';

describe('Test charity info componenet', () => {

  const info = {
    charityName: 'x',
    charityDescription: 'a real charity',
    charityWebsite: 'http://arealwebsite.com'
  }

  it('renders correctly', () => {
    const tree = renderer.create(
      <Charity info={info} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
