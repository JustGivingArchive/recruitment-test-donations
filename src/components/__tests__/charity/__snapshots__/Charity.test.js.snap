exports[`Test charity info componenet renders correctly 1`] = `
<div
  className="charity_container">
  <h2>
    Charity Name
  </h2>
  <p>
    Charity description
  </p>
  <img
    alt="logo of charity name"
    src="logo-url" />
</div>
`;
